doc/nrf/ug_thread_configuring.rst

```diff
@@ -23,40 +23,11 @@ Mandatory configuration
 To use the Thread protocol in |NCS|, set the following Kconfig options:
 
 * :option:`CONFIG_NET_L2_OPENTHREAD` - This option enables the OpenThread stack required for the correct operation of the Thread protocol and allows you to use them.
-* :option:`CONFIG_CPLUSPLUS` - This option must be enabled, because OpenThread is implemented in C++.
-* :option:`CONFIG_REBOOT` - This option is needed to ensure safe reboot.
-* :option:`CONFIG_ENTROPY_GENERATOR` - Required by both OpenThread and radio driver.
-* Options related to the Settings subsystem in the ``storage_partition`` partition of the internal flash:
-
-  * :option:`CONFIG_SETTINGS`
-  * :option:`CONFIG_FLASH`
-  * :option:`CONFIG_FLASH_PAGE_LAYOUT`
-  * :option:`CONFIG_FLASH_MAP`
-  * :option:`CONFIG_MPU_ALLOW_FLASH_WRITE`
-  * :option:`CONFIG_NVS`
 
 * General setting options related to network configuration:
 
   * :option:`CONFIG_NETWORKING`
-  * :option:`CONFIG_NET_UDP`
-  * :option:`CONFIG_NET_SOCKETS`
-
-IPv6 mandatory configuration
-============================
-
-The Thread protocol can only be used with IPv6.
-IPv4 is not supported.
-
-Enable the following options to make Thread work over IPv6:
-
-* :option:`CONFIG_NET_IPV6`
-* :option:`CONFIG_NET_CONFIG_NEED_IPV6`
-
-Additionally, since Thread by default registers a considerable amount of IP addresses, the default IPv6 address count values must be increased.
-Set the following options to the provided values:
-
-* :option:`CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT` to ``6``
-* :option:`CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT` to ``8``
+  * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread will only work in NCP architecture.
```

Comment:
```suggestion
  * :option:`CONFIG_NET_SOCKETS` - This option can be ommited if OpenThread is meant to work only in the NCP architecture.
```

Reply:
Also, `NCP architecture` can link to the NCP section from https://github.com/nrfconnect/sdk-nrf/pull/2605